### PR TITLE
MM-6950: restore last viewed on logout

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
These changes persist the last viewed team and channel to localStorage to be reused on login. The values persisted to local storage are keyed by user id to disambiguate multiple users using the same useragent.

I'm not super happy with these changes. A better approach would be to leverage the existing redux-persist subsystem, write these values to Redux, and have it manage the persistence. Unfortunately, on logout we `purge` everything redux-persist manages, so a solution to keep some values around would have to be crafted. One option might be to manually expunge keys, but then we'd want to be extra certain to have cleaned everything up. Another approach might be to persist different roots with whitelisted keys for the persistor that keeps data around after logout.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-6950

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: actions/admin_actions.jsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import * as AdminActions from 'mattermost-redux/actions/admin';
 import * as UserActions from 'mattermost-redux/actions/users';
 import {Client4} from 'mattermost-redux/client';
 
-import {clientLogout} from 'actions/global_actions.jsx';
+import {emitUserLoggedOutEvent} from 'actions/global_actions.jsx';
```
Comment (lieut-data) [inline actions/admin_actions.jsx]: `clientLogout` was basically `emitUserLoggedOutEvent` without the `Client4.logout` (odd name choice!), but this created multiple entry points for what essentially amounted to the same function. I've tested the various changes here and continue to get logged out.

File: actions/global_actions.jsx [hunk 2]
```diff
@@ -437,7 +441,13 @@ export function emitUserLoggedOutEvent(redirectTo = '/', shouldSignalLogout = tr
                 BrowserStore.signalLogout();
             }
 
-            clientLogout(redirectTo);
+            BrowserStore.clear();
```
Comment (lieut-data) [inline actions/global_actions.jsx]: No need to specify exclude keys here: the keys listed previously weren't managed by the browser store and lived on anyway. `selected_teams`, in particular, seems to have been entirely unused.

Comment (saturninoabril) [reply actions/global_actions.jsx]: My only concern is the RECENT_EMOJI_KEY.  So on logout then login, the recently used emojis are still there?

Comment (lieut-data) [reply actions/global_actions.jsx]: Correct: this threw me off for a while, believing I could influence what was preserved after logout, but it turns out `BrowserStore` doesn't manage keys outside of the `redux-persist` stuff. Recently used emojis -- alongside my new bits in the localStorage -- are preserved on logout.

File: tests/stores/local_storage_store.test.jsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        const channel2 = 'channel2';
+        const channel3 = 'channel3';
+        LocalStorageStore.setPreviousChannelName(userId1, teamId1, channel1);
+        LocalStorageStore.setPreviousChannelName(userId2, teamId1, channel2);
+        LocalStorageStore.setPreviousChannelName(userId2, teamId2, channel3);
+        assert.equal(LocalStorageStore.getPreviousChannelName(userId1, teamId1), channel1);
+        assert.equal(LocalStorageStore.getPreviousChannelName(userId2, teamId1), channel2);
+        assert.equal(LocalStorageStore.getPreviousChannelName(userId2, teamId2), channel3);
+    });
+});
```
Comment (saturninoabril) [inline tests/stores/local_storage_store.test.jsx]: 🎉

**Pull request merged**